File: ska_backend/src/server/route_api/route_assistant/route_ask_assistant_question/dto_ask_assistant_question.rs
use serde::{Deserialize, Serialize};

use crate::domain::user::dto_chat_details::DtoChatPacket;

#[derive(Clone, Serialize, Deserialize)]
pub struct AskAssistantQuestionRequest {
    pub chat_id: i64,
    pub question: String,
}

#[derive(Clone, Serialize, Deserialize)]
pub struct AskAssistantQuestionResponse {
    pub question: DtoChatPacket,
    pub answer: DtoChatPacket,
}

File: ska_frontend2/src/gui/page_assistant/dtos/dto_ask_assistant_question.ts
export class AskAssistantQuestionRequest {
    question: string;
    llm_model: string;
    prompt_template: string | null;

    constructor(obj: {
        question: string,
        llm_model: string,
        prompt_template: string | null,
    }) {
        this.question = obj.question;
        this.llm_model = obj.llm_model;
        this.prompt_template = obj.prompt_template;
    }
}

export class AskAssistantQuestionResponse {
    answer: string;

    constructor(obj: {
        answer: string
    }) {
        this.answer = obj.answer;
    }
}
